### PR TITLE
Blog on "Ensure secret pulled images" feature

### PR DESCRIPTION
**Pull request by sairameshv**
Place holder PR for a blog on the Ensure secret pulled images feature

Reference: https://github.com/kubernetes/enhancements/issues/2535 

/cc @haircommander

### CODE REVIEW

**Comment by rashansmith** [comment]
Hello, I'm Rashan from the 1.31 Release comms team. I'm reaching out with a reminder that the blog ready for review deadline is July 26, 2024. Please let me know how we can help!

**Comment by sairameshv** [comment]
> Hello, I'm Rashan from the 1.31 Release comms team. I'm reaching out with a reminder that the blog ready for review deadline is July 26, 2024. Please let me know how we can help!

Hey @rashansmith , Could you review the contents of this blog post and give some suggestions on it?

**Comment by rashansmith** [comment]
> > Hello, I'm Rashan from the 1.31 Release comms team. I'm reaching out with a reminder that the blog ready for review deadline is July 26, 2024. Please let me know how we can help!
> 
> Hey @rashansmith , Could you review the contents of this blog post and give some suggestions on it?

will do!

**Comment by sftim** [comment]
/hold

OK to unhold once Kubernetes v1.31 has been released.

File: content/en/blog/_posts/2024-07-01-ensure-secret-pulled-images.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+layout: blog
+title: "Kubernetes: 1.31 KubeletEnsureSecretPulledImages feature gate"
```
Comment (sftim) [inline content/en/blog/_posts/2024-07-01-ensure-secret-pulled-images.md:3]:
```suggestion
title: "Kubernetes 1.31: Additional Security Measures For Container Image Pulls"
```

**Review by sftim — COMMENTED** [review]
This looks like a [task](https://kubernetes.io/docs/tasks/) rather than post-release publicity.

Have a look at some of the other blog articles that we've gone out; I think you'll see they tend to take a different angle.

You can also link to the documentation about KubeletEnsureSecretPulledImages.

**Comment by sftim** [comment]
BTW @rashansmith the blog reviewers would typically review, but it's great if you're willing to help line up a reviewer.

This is open source so you are of course welcome to do your own reviewing.

File: content/en/blog/_posts/2024-07-01-ensure-secret-pulled-images.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+NAME                      READY   STATUS               RESTARTS   AGE
+image-with-valid-creds    1/1     Running              0          10s
+image-with-invalid-creds  1/1     ErrImageNotEnsured   0          10s
+```
+### Step 5: Repeat the pod creation after turning off the feature gate
+Modify the kubelet configuration by setting the `KubeletEnsureSecretPulledImages` feature gate to `false` and verify both the pods are `Running` successfully which is the default/current behavior.
+
+### Conclusion
+As of today, if a pod pulls an image from a private registry with valid credentials on to a node, the same can be leveraged by any another pod on the same node even without a valid image pull seceret/credentials.
+This newly introduced feature would help the cluster admin to have a better access control in terms of multi tenant scenarios by allowing the access of an image only incase of having a valid credentials even if the image is already present on the node.
```
Comment (T-Lakshmi) [inline content/en/blog/_posts/2024-07-01-ensure-secret-pulled-images.md:82]:
```suggestion
This newly introduced feature would help the cluster admin to have a better access control
in terms of multi tenant scenarios by allowing the access of an image only incase of having
a valid credentials even if the image is already present on the node.
```

**Comment by T-Lakshmi** [comment]
[Preview page](https://deploy-preview-47053--kubernetes-io-main-staging.netlify.app/blog/2024/07/01/ensure-secret-pulled-images/)

**Comment by divya-mohan0209** [comment]
@sairameshv / @sftim : Please may I know the status of this blogpost? Are we considering pushing it forward to a later date?

**Comment by sairameshv** [comment]
Hey @divya-mohan0209 , There is a new [PR](https://github.com/kubernetes/enhancements/pull/4789) for the [KEP](https://github.com/kubernetes/enhancements/issues/2535) proposing changes to this feature and that may require a new PR based on the latest behavior.
Hence, Closing this PR in favor of the same.

/close

**Pull request closed without merging**